app/vminsert/prompush/push.go

```diff
@@ -41,6 +43,13 @@ func Push(wr *prompb.WriteRequest) {
 		}
 		push(ctx, tssBlock)
 	}
+	if prommetadata.IsEnabled() && len(wr.Metadata) > 0 {
+		if err := ctx.WriteMetadata(wr.Metadata); err != nil {
+			logger.Errorf("cannot flush metrics metadata to storage: %s", err)
```

Comment:
Maybe specify that the error occurs when flushing **scraped** metrics metadata? Since errors from pushed data are returned directly to the client.